### ISSUE TITLE
Create a second base class to handle mixed spectrum models

### ISSUE THREAD

**Issue by lcwheeler** [BeerTheorySociety/ambigauss#13]

**Comment by jharman25** [comment]
Zach and I chatted about this earlier today - We're thinking we should write an entirely new BaseMixedSpectrum class that uses the same logic as BaseSpectrum but iterates over combinations of lineshapes. For example, if you have 2 peaks, it would iterate over fitting 1) two gaussians, 2) one gaussian and one lorentzian, and 3) two lorentzians. Then have some sort of scoring function to pick the best combination and spit that out as the best model. In the works!

**Comment by lcwheeler** [comment]
Makes sense to me! Maybe could do a likelihood ratio test between the different combos, or something similar?

**Comment by jharman25** [comment]
Sounds good!
